### DISCUSSION TITLE
Android application to use for OnlyKey challenge-response compatibility

### DISCUSSION THREAD

**Discussion by trippasnippa-420** [keepassxreboot/keepassxc#6578] [Q&A]
Hello,
I use and love KeepassXC on windows and linux and currently use KeepassDX on my android phone. I recently purchased an OnlyKey and would love to be able to use it to secure my database but I have to be able to somehow open the database on android. I have a USB OTG dongle for the phone and I understand that Keepass2Android supports YubiKey via NFC but I am looking for a OTG solution for the OnlyKey. The phone is rooted and has a full linux terminal installed so I can use terminal applications.

Any suggestions for an Android/Console app that can open KBDX4 databases and is compatible with the KeepassXC implementation of OnlyKey HMAC challenge-response?

**Comment by AlexiF** [comment]
Hello,
Did you find a solution?
I have the same issue: how to use a Yubkey in the easy way of KeePassXC on Android?
I can MFA with Keepass2Android in the same way that KeePass2 does: with an extra .xml file. But I would like to use KeePassXC...
https://keepassxc.org/docs/#faq-yubikey-no-extra-file

How to MFA with hardware key on KeePassXC and Android?

**Reply by trippasnippa-420** [reply]
I ended up settling with KeePass2Android, which supports USB security keys as long as you also have [ykdroid](https://github.com/pp3345/ykDroid) installed. For me I just select "Password + Challenge-Response for KeePassXC" in the drop down when opening a database and ykdroid pops up asking if I authorize the use of my security key.

Hope that helps :)

**Reply by AlexiF** [reply]
It helps! TY
